symengine/singleton.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import print_function, division
+from symengine import sympify
+
+ordering_of_classes = [
+    # singleton numbers
+    'Zero', 'One', 'Half', 'Infinity', 'NaN', 'NegativeOne', 'NegativeInfinity',
```

Comment:
I don't think we need separate classes. (It would be nice to have, but not necessary).
We just need to have `S(0) is S(0)`.